README

Package: Personal File Manager


* This package is available from https://github.com/ruittenb/p-f-m

* Prerequisites for this package, in dependency order:

                           at least      preferred
  perl                     5.8           5.14
    File::stat             1.00          1.05
      File::Stat::Bits     0.19          0.19 (*)
    File::Temp             0.22 ?        0.2304
    HTML::Parser           3.59 ?        3.72
      LWP                  5.827 ?       6.26
    Module::Load           0.16          0.32
  readline                 4.1           5
    Term-ReadLine-Gnu      1.09          1.24
  ncurses                  ?             5
    Term-Screen            1.06          1.06
      Term-ScreenColor     1.20          1.20


* Confusingly, the package File-Stat-Bits-1.01.tar.gz provides
  version 0.19 of the module File::Stat::Bits.

* If you are not sure if you have all the dependencies, you can run
  the 'install.sh' shell script. Otherwise, you can go straight to the
  'perl Makefile.PL; make; make test; sudo make install' phase.



Changes
2.12.1

* added: MIME types for .xz and .txz.
* added: shift-F8 for easy removal of oldmarks and newmarks.

2.12.0

* fixed: the si(Z)e command now can correctly figure out the exit status
  of the du(1) command.
* fixed: the radix for the (N)ame command was not correctly initialized in
  an initial swap state (started with the -s option).
* fixed: unnecessary delay when a Unix c(O)mmand fails.
* changed: the "Please Wait" message when reading a directory with many entries
  is now interruptible by pressing ESC.
* added: substitution modifier ={2/foo/bar}
* added: display countdown on "Please Wait" when a directory with a lot of
  entries is being read
* added: (M)ore - <F2> (Redescend): return to a previous child directory.
* added shift-F2 as alternative for (M)ore - <F2>.
* added: (M)ore - <F4> (Color) and (M)ore - <F9> (Layout) as alternatives for
  shift-F4 and shift-F9.

2.11.9

* fixed: a problem expanding sequences like '==8'.
* fixed: a problem expanding the '=7' escape.
* fixed: a problem when testing if a newer version is available.
* added: menu message providing help for the (N)ame command.
* changed: the semantics for the '=7' escape were changed.

2.11.8

* fixed: child processes were not always correctly reaped. This caused
  lots of zombie processes.
* fixed: if a child process took a long time to make new data available on
  the filehandle, the process was erroneously removed from the job stack.
* fixed: svn does not complain any more if the current directory is a working
  copy, but '..' is not.
* fixed: when writing bookmarks, bookmark 0 is now correctly saved.
* fixed: editing an ACL did not work if the fg_editor contained spaces.

* added MIME types to config file for: awk, sh, bash, csh, tcsh, zsh, shar,
  h, hh, lsp, scm, ss, rtx, sty, tcl, movie, spl, viv, vivo, vrml, wrl.
* added: ACL detection. If an ACL is present, a + is appended to the mode field.
* added helper app: hasacl(1) (currently Linux-only).
* added: (I)nclude - (D)irectories only
* added: (M)ore - (R)ead history/bookmarks prompts to read the input
  history and/or bookmarks from disk.
* changed: (M)ore - (W)rite history/bookmarks now prompts to write the
  input history and/or bookmarks to disk.

2.11.7

* renamed 'timestamptruncate' to 'timefieldstretch'
* added 'esc_timeout' option
* added '--login' option to start pfm as a login shell
* added: Directory keeps track of logical pathname so that the " command
  can switch back from physical to logical pathname
* added support for terminal keys 'ks4' (shift-F4) and 'ks9' (shift-F9)
  which cycle backward through colorsets and layouts, respectively.
* fixed problem where a newly created file in the current directory is not
  added if the current directory is not known by its canonical name

2.11.6-i18n

* added usage of Encode::Locale, to be able to correctly calculate
  localized string widths. Needs much testing.

2.11.6

* fixed: selecting the top file on the screen now correctly shows the
  mark on the file.
* fixed: the bookmark browser and Y-command browser don't accept clicks
  anywhere on the line any more, just on the item itself.
* fixed: a 'foreground editor' with spaces in the command definition will
  no longer give an error when editing the config file.
* fixed: swapping with <F7> now correctly restores the browser's view on
  the directory.
* fixed: creating a bookmark no longer gives an erroneous
  'Bookmark name not valid' message.
* added 'timestamptruncate' option to config file, which controls if the
  timestamp field will be adjusted to accommodate the complete timestamp
  string.
* added 'mouse_moves_cursor' option to config file, which controls if the
  mouse displaces the current file.
* added a file comparison information block which is displayed when the
  user is about to clobber a file.
* added 'clobber_compare' option to control whether to display the file
  comparison information.

2.11.5

* added: the bookmark browser and Y-command browser now apply highlighting
  to the current line
* fixed: refreshing the screen listing with RCS info now works again
  in the Your command browser
* fixed: The 'Y' character on the currentfile line is no longer missing
  in the Your command browser
* added several MIME types to the config file

2.11.4

* added support for numeric Your commands, e.g. (Y)our - (1)
* added support for numeric bookmarks, e.g. (M)ore - (G)o - (3)
* fixed bugs in screen resize and refresh in bookmark/Your command browser
* change in (M)ore - (G)o: only valid bookmarkkeys may report 'Bookmark not
  defined'. Keys that are not valid bookmarkkeys abort the action silently.

2.11.3

* first version with Your command browser
* fixed bug in bookmark browser: ^E and ^Y work again
* added icons to distribution

2.11.2

* added: support for Mercurial versioning system
* added: (M)ore - fo(L)low

2.11.1

* fixed quoting of '=p' escape and friends
* fixed crash when the 'defaultcolorset' was not defined as a colorset
* fixed bug positioning the cursor in the bookmark browser
* fixed bug positioning the cursor after (C)opy/(R)ename/(L)ink
* added commandline option '-c' (initial colorset)
* added =9 escape (previous path)
* added: if a file is (C)opied/(R)enamed/(L)inked to the previous directory
  (=9), it is added to its directory listing.
* reinstated 'checkforupdates' option. Call me indecisive.

2.11.0

* fixed bug with scrolling in bookmark browser
* fixed bug in handling mouse input in bookmark browser
* added 'footerkeys' to 'colorset' option
* deprecated 'mouseturnoff' option

2.10.9

* first version with bookmark browser
* fixed: suppress unnecessary svn error when statting '.svn'
* added 'menukeys' to 'colorset' option
* added 'cursorjumptime' option
* updated documentation

2.10.8

* fixed: no more crash in (C)opy/(R)ename/(L)ink when there is no swap state
* added: silent bookmark writing while testing

2.10.7

* fixed some small bugs

Makefile.PL
#!/usr/bin/env perl

# Makefile.PL for pfm

use ExtUtils::MakeMaker;

unless ($ENV{MAKEFILE_PL_CALLED_FROM_INSTALL_SH}) {
print <<_install_help_;

---------------------------------------------------------------
This Makefile will build, test and install pfm, and it will
rely on MakeMaker to install some of its dependencies.

Since this has proven to be unreliable at times, you are
advised to use the install script instead: 'sh install.sh'.

install.sh will install dependencies and re-run this Makefile.
---------------------------------------------------------------

_install_help_
}

WriteMakefile(
	NAME			=> 'pfm',
	VERSION_FROM		=> 'pfm',
	ABSTRACT		=> 'Personal File Manager for Unix/Linux',
	LICENSE			=> 'gpl',
	MIN_PERL_VERSION	=> 5.008,
	PREREQ_PM		=> {
		'File::stat'		=> 0,
		'File::Stat::Bits'	=> 0.19,   # maybe less
		'File::Temp'		=> 0.2304, # at least 0.22
		'Term::Cap'		=> 0,
		'Term::Screen'		=> 1.06,
		'Term::ScreenColor'	=> 1.20,
		'Term::ReadLine::Gnu'	=> 1.35,   # at least 1.16
		'HTML::Parser'		=> 3.72,   # at least 3.59
		'LWP'			=> 6.26,   # at least 5.827
		'Module::Load'          => 0.32,   # at least 0.16
	},
	EXE_FILES		=> ['pfm'],
#	DIR			=> ['listwhite'],
	MAN1PODS		=> {
		'pfm'			=> 'blib/man1/pfm.1',
#		'listwhite/listwhite.c'	=> 'blib/man1/listwhite.1',
	},
#	INSTALLSITESCRIPT
#	INST_SCRIPT
#	INST_BIN
);



